### PR TITLE
[LLDB][test] Provide a proper path to 'strip' utility for the LLDB API tests.

### PR DESCRIPTION
**Pull request by dzhidzhoev**
Use llvm-strip from llvm tools dir if no path to strip is set. Tests are changed to use cmake-provided strip.

This is useful for Windows host-Linux remote testing.

### CODE REVIEW

**Comment by DavidSpickett** [comment]
This seems fine, but what happens currently? Does it try to use a tool that's built for the remote not the host?

**Review by labath — COMMENTED** [review]
Could we standardize on `$(OBJCOPY) --strip-debug` instead?

**Review by JDevlieghere — CHANGES_REQUESTED** [review]
Please don't use environment variables to pass options to the `make` invocation. It makes it really hard to reproduce build failures because you can't just copy-paste the failing invocation to debug the build. I went through an effort to clean that up a few years ago. If there are other instances that do this then they slipped in without my knowledge. 

A similar problem applies to the `dotest.py` invocation and the `lldb-dotest.py` wrapper configured by CMake that a bunch of folks rely on. 

TL;DR: instead of an environment variable, please add this as an option to `dotest.py` and pass it explicitly as part of the `make` invocation in `builder.py`. You can look at `dsymutil` for inspiration.

**Comment by dzhidzhoev** [comment]
> Please don't use environment variables to pass options to the `make` invocation. It makes it really hard to reproduce build failures because you can't just copy-paste the failing invocation to debug the build. I went through an effort to clean that up a few years ago. If there are other instances that do this then they slipped in without my knowledge.
> 
> A similar problem applies to the `dotest.py` invocation and the `lldb-dotest.py` wrapper configured by CMake that a bunch of folks rely on.
> 
> TL;DR: instead of an environment variable, please add this as an option to `dotest.py` and pass it explicitly as part of the `make` invocation in `builder.py`. You can look at `dsymutil` for inspiration.

Thank you!



> This seems fine, but what happens currently? Does it try to use a tool that's built for the remote not the host?

Currently, it's not found on Windows and I see a couple of tests that are disabled because of that (e.g. https://github.com/llvm/llvm-project/blob/7b99b1d4f733ed5e1b206f0b392b0864e7a0d468/lldb/test/API/functionalities/json/symbol-file/TestSymbolFileJSON.py#L16). Yes, it can be installed externally added to PATH. But also llvm-strip from LLVM_TOOLS_BINARY_DIR can be utilized to reduce dependencies.

I'll try the solution proposed by @labath and replace strip usage with objcopy.
